### ISSUE TITLE
[ENH] Support for discrete output distributions and probabilistic classification

### ISSUE THREAD

**Issue by fkiraly** [sktime/skpro#1003]
Even though `scikit-learn` supports probabilistic classification and, thus, discrete output distributions, it has a few problems:
* the `predict_proba` API is inconsistent with the `skpro` regression API, as it returns a numpy array and not a distribution object
* there is no API for multivariate probabilistic classification, or ordinal regression.

It was suggested by @felipeangelimvieira to add a `skpro` native probabilistic classifier API that can represent a wider range of return distributions.

Someone working on this should give more details about the API design first.

As a starting point, I suggest:

* a new module `classification`
* a base class `BaseProbaClassifier` with methods `fit`, `predict` and `predict_proba`
* if ordinal classification is also covered, then `predict_quantiles` and `predict_interval` may also make sense, in this case a capability tag for ordinal classification
* a distribution `DiscreteClass` used for the output
* a `scikit-learn` adapter that allows to expose any sklearn probabilistic classifier under the modified API

**Comment by ANANYA542** [comment]
Hi @fkiraly, I'd like to work on this I went through  BaseProbaRegressor,  BaseSurvReg, and the BaseDistribution hierarchy to align with existing patterns. Here's my proposed design:
**Module layout:**
skpro/classification/
├── base/_base.py          # BaseProbaClassifier
├── adapters/_sklearn.py   # SklearnProbaClassifier
└── tests/
skpro/distributions/categorical.py   # new distribution

BaseProbaClassifier — extends BaseEstimator directly (not  BaseProbaRegressor, since outputs differ fundamentally):
1. fit(X, y) → stores classes_ from training labels
2. predict(X) → pd.DataFrame of predicted labels (default: mode of  predict_proba)
3. predict_proba(X) → Categorical distribution object, not a numpy array
Optional predict_interval / predict_quantiles for ordinal classification, gated behind a capability:ordinal tag — but I'd defer this to a follow-up.

Categorical distribution — distr:measuretype = "discrete", parameterized by p (n_samples × n_classes) and classes:
1. pmf(x) — vectorized per-sample probability for given class labels
2. sample() — draws class labels according to stored probabilities
3. mean() — well-defined only for numeric classes (sum(p_i * class_i)); for non-numeric, would raise or fall back to mode. Happy to discuss the right convention here.

SklearnProbaClassifier adapter:
clf = SklearnProbaClassifier(RandomForestClassifier())
clf.fit(X_train, y_train)
d = clf.predict_proba(X_test)  # → Categorical distribution
d.pmf("class_A")               # per-sample P(class_A)

PR plan:

1. Categorical distribution + tests
2. BaseProbaClassifier + sklearn adapter + tests
3. (follow-up) ordinal classification support

One naming question: the issue mentions DiscreteClass — I used Categorical to match scipy/PyTorch/TF conventions, but happy to rename. Would love your feedback on the design

**Comment by Ironankit525** [comment]
i would like to work on it,can you plz assign it to me.

**Comment by Shikhar-Kesharwani** [comment]
Hi @fkiraly,

I've reviewed `skpro`'s `BaseDistribution` and `BaseProbaRegressor` architecture to explore API design options for probabilistic classification.

While `skpro` currently supports discrete distributions such as `Poisson` and `Binomial`, these rely on numeric integer supports. A native nominal/categorical distribution over arbitrary class labels appears to be missing.

To resolve this cleanly, I propose a two-stage direction:

1. **`Categorical` distribution:** First introduce a standalone `Categorical` distribution in `skpro.distributions` supporting `pmf`, `log_pmf`, `mode`, and `sample`. For tabular data, probabilities could potentially be represented using an instance-by-class DataFrame while preserving instance indexing. Does this convention align with your architectural expectations, or would you prefer an alternative representation?

2. **Classification module:** Once the distribution contract is established, introduce `BaseProbaClassifier` and an sklearn adapter, with the proposed `predict_proba` interface returning a `Categorical` distribution object.

Regarding PR #1029, I noticed it attempted the broader scope but has been inactive for several months. To avoid duplicating or conflicting with prior work, do you consider #1029 stalled, and would a modular redesign be welcome?

Does this two-stage approach look directionally correct, and should I prepare a more detailed API sketch for your review? I would wait for your feedback on the design before implementing it.
